### ISSUE TITLE
Ellie stuck "compiling" with some libraries

### ISSUE THREAD

**Issue by joakin** [ellie-app/ellie#74]
Thanks for Ellie, it is great 🙇‍♂️.

I was trying to make a new ellie with the updated version of my library to link to it from the library docs, and I'm stuck. Previous versions of the library worked fine, but for some reason the one I released a few days ago doesn't.

## Steps

1. https://ellie-app.com/new
1. Click the *packages* icon to add a library
1. Search for "joakin/elm-canvas"
1. Click "Install" on "elm-canvas - joakin - 3.0.0"
1. Click "X" to close the packages view
1. Click "Compile"

**Expected**

* After some time, the default example is compiled and shows in the view on the right

**Actual**

* Ellie is stuck "Compiling" (button on the top), and the green power socket icon on the bottom right turns red after a bit.

This happens regardless of browser, making a new ellie, and also using the namespace from the imported library.

Tried on macOS 10.14.1 (18B75) with:
* Firefox 64
* Chrome 71

Both in incognito/private mode (no extensions).

I saved an example that doesn't work here: https://ellie-app.com/4bmG7S4bH6ba1

Let me know if there is anything else I can do to help. I've tried the library locally with elm-make and install and had no problems, so I think it is an issue with the website.

Thanks!

**Comment by joakin** [comment]
I published 3.0.1 with a small README change to see if that would fix it, but sadly it still gets stuck :(

**Comment by ethanlee16** [comment]
Hi! I just tried installing your package on Ellie and ran into the same issue. I'm looking into debugging right now, but right now what I've got is:

`{"type":"error","path":"elm.json","title":"MISSING DEPENDENCIES","message":["Your elm.json is missing some \"indirect\" dependencies:\n\n    ",{"bold":false,"underline":false,"color":"yellow","string":"\"avh4/elm-color\": \"1.0.0\"\n    \"elm/json\": \"1.1.2\"\n\"elm/time\": \"1.0.0\"\n    \"elm/url\": \"1.0.0\"\n    \"elm/virtual-dom\": \"1.0.2\""},"\n\nThis usually means you are editing elm.json by hand. It is much more reliable to\nuse the ",{"bold":false,"underline":false,"color":"GREEN","string":"elm install"}," command instead."]}
`

Hopefully this helps if anyone watching is a regular contributor - I will be looking into it but may take me a bit more time.

**Comment by tbash** [comment]
Confirmed this is not an issue due to the Now config, https://ellie.tba.sh/4h8cpKDYQqxa1 is deployed to an AWS infrastructure and is suffers the same fate.

**Comment by joakin** [comment]
That is very strange. The library installs and works fine in OSX 10.14.

I've made a small example [sketch repo](https://github.com/joakin/elm-canvas-sketch) in case you want to reproduce.

Thanks for looking into this 💟

**Comment by tbash** [comment]
@joakin I looked into this a little more this evening, it would appear this is an issue isolated to at least some (if not all) Linux distributions and attempting to install https://github.com/joakin/elm-canvas.

I have attempted to do so on Ubuntu and CentOS, both yield the same results when running:
```sh
$ elm install joakin/elm-canvas
```
With the output:
```sh
Dependencies ready!
-- CORRUPT DEPENDENCY ----------------------------------------------------------

I ran into a problem while building the following package:

    joakin/elm-canvas 3.0.1

This probably means the downloaded files got corrupted somehow. Try deleting
/home/ec2-user/.elm (a directory for caching build artifacts) and see if that
resolves the issue.
```

I'll leave this open for a bit while we figure out how to report this to https://github.com/elm/compiler but this is not an issue specific to Ellie, only a side-effect of the OS it is distributed on. An SSCCE may not even be necessary as running the following is enough to recreate this problem (after agreeing to the prompts with `y`).
```sh
$ elm init
$ elm install joakin/elm-canvas
```

**Comment by prasmussen** [comment]
@joakin I'm not sure how the elm compiler builds the filepath to a module, but you might need to have a uppercase `src/Canvas` directory. OSX has a case insensitive filesystem by default, but on Linux the default is to be case sensitive which might explain why it works on OSX with the lowercase directory.

**Comment by joakin** [comment]
Wow @prasmussen that was it! How did you notice?

I published a new version with the rename, 3.0.2, and now it installs and compiles fine on ellie-app:

* https://ellie-app.com/4jxw7rnQ9q6a1

I feel stupid, and then again, the install error of *CORRUPT DEPENDENCY* is so obscure...

Thank you all for chiming in, I'm so happy I could update the ellie and docs for v3 finally 🙇

**Comment by joakin** [comment]
I've opened an issue in elm/compiler: https://github.com/elm/compiler/issues/1875

I'd appreciate any feedback since you are more experienced on the ecosystem and did reproduce the issues.

I hope this helps fix this problem, thank you all.
